frontend/src/utilities/useWatchIntegrationComponents.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import { cloneDeep } from 'lodash-es';
+import { useAppSelector } from '~/redux/hooks';
+import { OdhApplication } from '~/types';
+import { getIntegrationAppEnablementStatus } from '~/services/integrationAppService';
+import { POLL_INTERVAL } from './const';
+import { isIntegrationApp } from './utils';
+
+export const useWatchIntegrationComponents = (
+  components: OdhApplication[],
+): { checkedComponents: OdhApplication[]; isIntegrationComponentsChecked: boolean } => {
+  const [isIntegrationComponentsChecked, setIsIntegrationComponentsChecked] = React.useState(false);
+  const forceUpdate = useAppSelector((state) => state.forceComponentsUpdate);
+  const initForce = React.useRef<number>(forceUpdate);
+  const integrationComponents = React.useMemo(
+    () => components.filter((component) => isIntegrationApp(component)),
+    [components],
+  );
+  const [newComponents, setNewComponents] = React.useState<OdhApplication[]>([]);
+
+  const updateComponentEnablementStatus = (
+    integrationComponentList: OdhApplication[],
+    componentList: OdhApplication[],
+  ) => {
+    const newComponentsList = cloneDeep(componentList);
+
+    integrationComponentList.forEach((component) => {
+      if (component.spec.internalRoute) {
+        getIntegrationAppEnablementStatus(component.spec.internalRoute).then((response) => {
+          if (response.error) {
+            setNewComponents(newComponentsList);
+          } else {
+            newComponentsList[componentList.indexOf(component)].spec.isEnabled =
+              response.isAppEnabled;
+            setNewComponents(newComponentsList);
```

Comment:
What happens if `component` is not found? 

Maybe map over the results and supply a new value?

I think you want something like this:
```suggestion
            setNewComponents(
              componentList.map((app) =>
                app.metadata.name === component.metadata.name
                  ? {
                      ...app,
                      spec: {
                        ...app.spec,
                        isAppEnabled: response.isAppEnabled,
                      },
                    }
                  : app,
              ),
            );
```

Reply:
updated